File: Examples/CSharp/Text/ExtractingAllText.cs

using System.IO;
using Aspose.Note;
using System;
using System.Linq;

namespace Aspose.Note.Examples.CSharp.Text
{
    public class ExtractingAllText
    {
        public static void Run()
        {
            // ExStart:ExtractingAllText
            // ExFor:Page.GetChildNodes
            // ExFor:RichText
            // ExFor:RichText.Text
            // ExSummary:Shows how to get all text from the document.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Text();

            // Load the document into Aspose.Note.
            Document oneFile = new Document(dataDir + "Aspose.one");

            // Retrieve text
            string text = string.Join(Environment.NewLine, oneFile.GetChildNodes<RichText>().Select(e => e.Text)) + Environment.NewLine;

            // Print text on the output screen
            Console.WriteLine(text);

            // ExEnd:ExtractingAllText
        }
    }
}

File: Examples/VisualBasic/Text/ExtractingAllText.vb
Imports System.IO
Imports Aspose.Note

Namespace Text
    Public Class ExtractingAllText
        Public Shared Sub Run()
            ' ExStart:ExtractingAllText
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Text()

            ' Load the document into Aspose.Note.
            Dim oneFile As New Document(dataDir & Convert.ToString("Aspose.one"))

            ' Retrieve text
            Dim text As String = oneFile.GetText()

            ' Print text on the output screen
            Console.WriteLine(text)
            ' ExEnd:ExtractingAllText
        End Sub
    End Class
End Namespace